### PR TITLE
Hobbit

### PR DESCRIPTION
**Pull request by pl608**
Allows external mods to make changes to the player model
don't know if this will work on more advanced models
also don't know if this was necessary however it works and that is all I was shooting for :)
- [x] This PR has been tested locally

### CODE REVIEW

**Comment by LoneWolfHT** [comment]
Just have your mod edit the animations in `player_api.registered_models["character.b3d"]` after depending on the `ctf_player` mod, no need for a setting or completely replacing the CTF model

**Comment by pl608** [comment]
ok. Though the classes update that I added will be needed since otherwise it will be a normal sized player anytime someone is a knight or scout (and potentially medic too)

File: mods/ctf/ctf_modes/ctf_mode_classes/classes.lua [hunk 1]
```diff
@@ -12,7 +12,7 @@ local class_props = {
 		color = "grey",
 		description = "High HP class with a sword capable of short damage bursts",
 		hp_max = 30,
-		visual_size = vector.new(1.1, 1.05, 1.1),
+		visual_size =  vector.new(0.1, 0.05, 0.1),
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_modes/ctf_mode_classes/classes.lua:15]:
```suggestion
		visual_size = vector.new(0.1, 0.05, 0.1),
```

File: mods/ctf/ctf_modes/ctf_mode_classes/classes.lua [hunk 2]
```diff
@@ -47,7 +47,7 @@ local class_props = {
 		name = "Scout",
 		color = "orange",
 		description = "Ranged class with a scoped rifle/grenade launcher and a scaling ladder for reaching high places",
-		visual_size = vector.new(0.9, 1, 0.9),
+		visual_size =  vector.new(-0.1, 0.0, -0.1),
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_modes/ctf_mode_classes/classes.lua:50]:
```suggestion
		visual_size = vector.new(-0.1, 0.0, -0.1),
```

File: mods/ctf/ctf_modes/ctf_mode_classes/classes.lua [hunk 3]
```diff
@@ -375,10 +375,10 @@ end
 
 function classes.update(player)
 	local class = classes.get(player)
-
+	local base_size = vector.new(player_api.registered_models['character.b3d'].visual_size)
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_modes/ctf_mode_classes/classes.lua:378]: Why is vector.new() used here?

File: mods/ctf/ctf_modes/ctf_mode_classes/classes.lua [hunk 4]
```diff
@@ -375,10 +375,10 @@ end
...
 	player:set_properties({
 		hp_max = class.hp_max or minetest.PLAYER_MAX_HP_DEFAULT,
-		visual_size = class.visual_size or vector.new(1, 1, 1)
+		visual_size = vector.add(base_size,class.visual_size ) or base_size
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_modes/ctf_mode_classes/classes.lua:381]:
```suggestion
		visual_size = vector.add(base_size, class.visual_size) or base_size
```

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]

**Comment by LoneWolfHT** [comment]
You can just commit all those suggestions, I put them there so you can see how I prefer code style

File: mods/ctf/ctf_modes/ctf_mode_classes/classes.lua (hunk 3 above)
Comment (pl608) [reply mods/ctf/ctf_modes/ctf_mode_classes/classes.lua:378]: because it was used somewhere else and I didn't want to break anything

**Comment by LoneWolfHT** [comment]
@pl608 Instant crash on server start, I thought you tested this locally?

**Comment by pl608** [comment]
what was the error? I never had an issue with and without my mod

**Pull request merged**